impress-org/givewp
# Fix: remove orphan tables when deleting a sub-site

Resolves [GIVE-119]

## Description

@Genevieve-K mentioned in https://github.com/impress-org/givewp/pull/7120#discussion_r1414247185 that when removing a sub-site in a multisite installation, the Give tables are not being deleted. Consequently, when a new sub-site with the same ID is added, the orphan data from the previous sub-site becomes part of the new sub-site.

Upon investigation, I discovered that we already have an action dropping tables upon sub-site deletion. However, it was using an outdated list of custom tables. This pull request refactors that action with a different approach to programmatically list all custom tables linked to that sub-site instead of having a hardcoded list of tables.

However, there is an important requirement for that action to be called: the Give plugin must be activated either for the entire network or at least in the main site. Otherwise, it will not detect when a sub-site is removed.

## Affects

Multisite

## Testing Instructions

1. Activate Give for the entire network or at least for the main site
2. Create a new sub-site
3. Activate Give on it, if not yet activated for the network
4. Check that Give tables were created
5. Remove the recently created sub-site
6. Confirm all Give tables for that sub-site were removed

## Pre-review Checklist






[GIVE-119]: https://stellarwp.atlassian.net/browse/GIVE-119?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
